### ISSUE TITLE
Linker Error?

### ISSUE THREAD

**Issue by ianfmc** [CocoaPods/CocoaPods#3155]
I have a very simple podfile, and processed using Cocoapods v0.35.0. 

```
platform :ios, '7.1'

source 'https://github.com/CocoaPods/Specs.git'

link_with ['RewindPolo', 'RewindPoloTests']

pod 'MagicalRecord', :git => 'https://github.com/magicalpanda/MagicalRecord.git', :bra\
nch => 'develop'
pod 'OCMock'
```

When I try to run the Unit Test in XCode 6.0, I get the error:
```
ld: library not found for -lPods-RewindPolo'
```
Where 'RewindPolo' is the name of the application. I tried making specific targets for the two targets ('RewindPolo' and 'RewindPoloTests') but got a different error (from **pod install**) in that case (related to the configuration -- 
```
CocoaPods did not set the base configuration of your project because your project already has a custom config set
```

I reviewed issue #2752, and and #3099, but was not able to resolve using the solutions described there.

**Comment by ianfmc** [comment]
here is the complete error

```
Ld /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphoneos/rewindpolonew.app/rewindpolonew normal armv7
    cd /Users/ianmccal/Development/RewindPolo/RewindPolo
    export IPHONEOS_DEPLOYMENT_TARGET=7.1
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch armv7 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.1.sdk -L/Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphoneos -F/Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphoneos -F/Users/ianmccal/Development/RewindPolo/RewindPolo -filelist /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Intermediates/RewindPolo.build/Debug-iphoneos/RewindPolo.build/Objects-normal/armv7/rewindpolonew.LinkFileList -dead_strip -ObjC -lPods-MagicalRecord -framework CoreData -fobjc-arc -fobjc-link-runtime -miphoneos-version-min=7.1 -lPods -lPods-RewindPolo -lsqlite3 -lz -framework SystemConfiguration -framework Security -framework QuartzCore -framework MobileCoreServices -framework CoreLocation -framework CFNetwork -framework AudioToolbox -framework MessageUI -framework CoreGraphics -framework UIKit -framework Foundation -framework Parse -framework Bolts -Xlinker -dependency_info -Xlinker /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Intermediates/RewindPolo.build/Debug-iphoneos/RewindPolo.build/Objects-normal/armv7/rewindpolonew_dependency_info.dat -o /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphoneos/rewindpolonew.app/rewindpolonew
```

**Comment by ianfmc** [comment]
Continuing to experiment, I deleted the DerivedData directories, and changed the Podfile to use only OCMock (since that is not pulling from a **develop** branch). So the Podfile looks like

```
# Uncomment this line to define a global platform for your project
platform :ios, '6.0'

pod 'OCMock'
```

Not sure if that could be simpler. The error is:

```
d /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator/RewindPoloTests.xctest/RewindPoloTests normal x86_64
    cd /Users/ianmccal/Development/RewindPolo/RewindPolo
    export IPHONEOS_DEPLOYMENT_TARGET=7.1
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -bundle -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk -L/Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator -F/Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk/Developer/Library/Frameworks -F/Applications/Xcode.app/Contents/Developer/Library/Frameworks -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk/Developer/Library/Frameworks -filelist /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Intermediates/RewindPolo.build/Debug-iphonesimulator/RewindPoloTests.build/Objects-normal/x86_64/RewindPoloTests.LinkFileList -bundle_loader /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator/RewindPolo.app/RewindPolo -Xlinker -objc_abi_version -Xlinker 2 -ObjC -lPods-OCMock -fobjc-arc -fobjc-link-runtime -Xlinker -no_implicit_dylibs -mios-simulator-version-min=7.1 -framework XCTest -framework UIKit -framework Foundation -Xlinker -dependency_info -Xlinker /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Intermediates/RewindPolo.build/Debug-iphonesimulator/RewindPoloTests.build/Objects-normal/x86_64/RewindPoloTests_dependency_info.dat -o /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator/RewindPoloTests.xctest/RewindPoloTests

ld: file not found: /Users/ianmccal/Library/Developer/Xcode/DerivedData/RewindPolo-cfzfensoeeeeiqhikvxghmgivyqa/Build/Products/Debug-iphonesimulator/RewindPolo.app/RewindPolo
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by ianfmc** [comment]
Final (perhaps very important) note for this evening: this only occurs when I try to Test (cmd-U)

**Comment by ianfmc** [comment]
Am very perplexed: created the same Podfile for a simple test project and it works.

**Comment by ianfmc** [comment]
OK...figured out what was going on.

I had to change the bundle name (after moving from **Testflightapp** to **iTunes Connect**), which I did by hard-coding the product name. In that case, the test target bundle name was now different (because it  was being generated automatically from the Product name). I finally put it all together finally, and realized that I need to hard code bundle name, and once I did that, I could both upload to iTunes **AND** Unit Test. 

With more people having to switch to iTunes connect, this may arise more and more, so I wanted to get the resolution out there.
